runtime/src/gpu/rocm/gpu-rocm.c

```diff
@@ -451,7 +451,7 @@ void* chpl_gpu_mem_array_alloc(size_t size, chpl_mem_descInt_t description,
   // (chpl_gpu_device_alloc == true during this time), in this case the intent
   // is just to return host memory anyway so there's no need to switch
   // contexts.
-  if(!chpl_gpu_device_alloc) { return chpl_malloc(size); }
+  if(!chpl_gpu_device_alloc) { return chpl_mem_alloc(size, description, lineno, filename); }
```

Comment:
You've explained this to me, but I forgot and it might be good to have it in writing: I can't relate how this works around the fact that context logic not working with multiple devices.

I also don't remember how that wasn't working, tbh. We did use multiple AMD gpus in a node before?

Reply:
Whatever the issue was I'm unable to reproduce it so I'm removing that code and its comment. I do see several `AOD` failures but they don't seem related to it (the failure repros whether or not I have that line).